homeassistant/components/acmeda/sensor.py

```diff
@@ -0,0 +1,239 @@
+        #        group_progress,
+        #    )
+        # )
+
+        await asyncio.wait(tasks)
+
+    await update_hub()
+
+
+async def async_update_items(
+    hass,
+    config_entry,
+    hub,
+    async_add_entities,
+    request_hub_update,
+    is_group,
+    current,
+    progress_waiting,
+):
+    """Update covers from the hub."""
+    if not hub.authorized:
+        return
+
+    if is_group:
+        api_type = "group"
+        api = hub.api.rooms
+    else:
+        api_type = "roller"
+        api = hub.api.rollers
+
+    try:
+        start = monotonic()
+        with async_timeout.timeout(8):
+            await hub.async_request_call(hub.api.update())
+            await hub.api.event_update.wait()
+    except (asyncio.TimeoutError) as err:
+        _LOGGER.debug("Failed to fetch %s: %s", api_type, err)
+
+        if not hub.available:
+            return
+
+        _LOGGER.error("Unable to reach hub %s (%s)", hub.host, err)
+        hub.available = False
+
+        for item_id, item in current.items():
+            if item_id not in progress_waiting:
+                item.async_schedule_update_ha_state()
+
+        return
+
+    finally:
+        _LOGGER.debug(
+            "Finished %s request in %.3f seconds", api_type, monotonic() - start
+        )
+
+    if not hub.available:
+        _LOGGER.info("Reconnected to hub %s", hub.host)
+        hub.available = True
+
+    new_items = []
+
+    for item_id, item in api.items():
+        if item_id not in current:
+            current[item_id] = AcmedaBattery(hass, item, hub, is_group)
+
+            new_items.append(current[item_id])
+        elif item_id not in progress_waiting:
+            current[item_id].async_schedule_update_ha_state()
+
+    await remove_devices(hass, config_entry, api, current)
+
+    if new_items:
+        async_add_entities(new_items)
+
+
+class AcmedaBattery(AcmedaBase, entity.Entity):
+    """Representation of a Acmeda cover device."""
+
+    device_class = DEVICE_CLASS_BATTERY
+    unit_of_measurement = UNIT_PERCENTAGE
+
+    def __init__(
+        self, hass, roller: aiopulse.Roller, hub: aiopulse.Hub, is_group=False
+    ):
+        """Initialize the roller."""
+        super().__init__(hass, roller, hub)
+
+    @property
+    def name(self):
+        """Return the name of roller."""
+        return super().name + " Battery"
+
+    @property
+    def battery_level(self):
+        """Return the battery level of the device."""
+        return self.roller.battery
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self.roller.battery
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attr = {}
+        super_attr = super().device_state_attributes
+        if super_attr is not None:
+            attr.update(super_attr)
+
+        attr[ATTR_BATTERY_LEVEL] = self.roller.battery
+        attr[ATTR_BATTERY_CHARGING] = False
```

Comment:
```suggestion
        attr[ATTR_BATTERY_CHARGING] = False
```
If this doesn't support charging, please remove.